src/java.desktop/share/classes/javax/swing/plaf/basic/BasicScrollBarUI.java

```diff
@@ -1613,7 +1614,15 @@ public void actionPerformed(ActionEvent e) {
             // and mouseReleased is not called, then timer will not be stopped
             // Stop the timer if frame is disabled
             Component parent = scrollbar.getParent();
+            Component focusOwner = FocusManager.getCurrentManager().getFocusOwner();
             do {
+                // If application isn't in focus, stop the timer
+                if (focusOwner == null) {
```

Comment:
```suggestion
            do {
                if (!scrollbar.isShowing()) {
```

I haven't tested this solution well, but you can try to stop the timer when the scrollbar is not visible.

Reply:
Tested this solution locally and in CI. Seems to work well. Clientlibs passes and the updated, automatic test also passes on all related platforms. Updated the PR. Thanks for providing the automated test and suggested change.

Reply:
I agree, `!scrollbar.isShowing()` is a better condition to check for.

Reply:
@DamonGuy Probably a good idea to add @azvegint as co-author to give credit for the fix and automated test.

Reply:
Yep! Good reminder. Was going to do so.